lib/Reader/reader.cpp

```diff
@@ -6994,6 +7001,7 @@ void ReaderBase::readBytesForFlowGraphNodeHelper(
     case ReaderBaseNS::CEE_LDC_I4_8:
     case ReaderBaseNS::CEE_LDC_I4_M1:
     LOAD_CONSTANT:
+      nop();
```

Comment:
Ah, I think I see what you're getting at ...certain MSIL operations generate operands, not instructions. So if we have an MSIL operation that only creates an operand, and it has a reportable debug location, and the following MSIL operation creates an instruction and has some other reportable debug location, that first debug location doesn't get recorded anywhere and is lost.

So arguably we only need this `nop` if we're at a stack empty point, right? And presumably `ldc` is the only case where we generate an operand and not an instruction?

You might be able to make this more robust in the main IL processing loop -- if you are at a stack empty, down at the bottom of the loop after we've generated the IR, if the last instruction in the current block doesn't have the stack empty debug info, you could generate a `nop` there. But if `ldc` is truly the only case of this then adding a comment here (and possibly only adding a `nop`\ for stack empty) is ok by me.